first stab at adding syntax checking on save similar to GoSublime

Jayflux opened the pull request:
HI guys
I really liked the syntax checking on save within GoSublime.
So i had a go at trying it for Rust.

Under the hood all this plugin does is call "cargo run" in the same directory the file is in, if there's any errors it will try and parse out the first line number, then return that back to sublime.

It would be nice if some sublime users could take a look at my PR and tell me what they think needs adding or their thoughts.

here is a demo
Thanks Jason

![testingpr](https://cloud.githubusercontent.com/assets/936006/15608781/fd8485d8-2413-11e6-8dcf-ead623661278.gif)

Jayflux commented:
changing cargo run to cargo rustc -- -Zno-trans
however with very large projects this seems to fail at updating the registry, if anyone has any idea let me know!

Jayflux commented:
Fixed the error now, and added async so no hanging now either!

Jayflux commented:
Hi all, its now faster and....
I've added tooltips, check out the new gif:

![testingpr2](https://cloud.githubusercontent.com/assets/936006/15618044/e7e5a614-2443-11e6-99c7-413c4881999e.gif)

steveklabnik commented:
Neat! I don't use sublime personally, so I'm not sure I want to merge.

Jayflux commented:
It would be good to hear some reviews from people who use sublime to see what they think

Aaronepower commented:
@Jayflux It looks great! My only question is why do all tooltips start with a `'b`?

Jayflux commented:
Its a bug, i'm working on that

Aaronepower commented on plugin.py:
```diff
@@ -0,0 +1,78 @@
+    return original_pt
+
+
+def lol():
```
I assume this was unintentionally left in?

Jayflux commented:
@Aaronepower the tooltip bug if fixed, here's a preview
It looks like a callback has to be provided to show_popup_menu so ive just created a callback function which does nothing.

![testingpr3](https://cloud.githubusercontent.com/assets/936006/15657328/b90c2636-26a7-11e6-8c35-ff6dcd880bac.gif)

Aaronepower commented:
@Jayflux Looking great. You should probably add a comment on the callback explaining that behaviour, and have you tested this against projects with a build script? As they typically put all the code into `main.rs` or `lib.rs`. I don't think it's necessary to cover this case, just to make sure they don't crash the plugin.

[tokei](https://github.com/Aaronepower/tokei) uses a build script, so you can test it on that if you need.

Jayflux commented:
Just tested on tokei
Because cargo reports the error is in main.rs and the actual error was in marcos.rs the plugin failed to show a dot on marcos.rs

We decided this isn't really an issue and there's very little we can do about it, the plugin just silently fails

Jayflux commented:
@Aaronepower I've now added settings, so its not on by default.
I've also updated the readme on how to add settings and how to add syntax checking.
https://github.com/Jayflux/sublime-rust/tree/addingSyntaxChecking

Jayflux commented:
@steveklabnik could you take a look at this?
There's some more improvements I can add, but I think i'm done (for now)

steveklabnik commented:
As I said above, it seems good to me, but I also don't use sublime. I'm not actually sure who is in charge of the sublime plugin. Looking at recent PRs, it looks like @brson and myself were the last ones to merge things. He is on vacation, so I'm gonna just merge this. :+1:

dten commented:
Don't want to be a spoilsport but there's already a pretty good linter here.
https://github.com/oschwald/SublimeLinter-contrib-rustc
